src/Kiota.Builder/Configuration/GenerationConfiguration.cs

```diff
@@ -95,6 +97,14 @@ public bool ClearCache
         get; set;
     }
     public HashSet<string> DisabledValidationRules { get; set; } = new(0, StringComparer.OrdinalIgnoreCase);
+    public int MaxDegreeOfParallelism { get; set; } = -1;
+    public GenerationConfiguration()
+    {
+        if (Int32.TryParse(Environment.GetEnvironmentVariable("KIOTA_GENERATION_MDOP"), out var nrThreads))
+        {
+            MaxDegreeOfParallelism = nrThreads;
+        }
+    }
```

Comment:
```suggestion
```

Again, parsing the environment variable is already done by the IConfigurationProvider. No need to duplicate.